core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1192,12 +1201,13 @@ write_files:
                       # Fetch the list of instances being terminated by their respective ASGs
                       instances_to_drain=$(asg describe-auto-scaling-groups | jq -r '.AutoScalingGroups[] | select((.Tags[].Key | contains("kube-aws")) and (.Tags[].Key | contains("kubernetes.io/cluster/{{.ClusterName}}"))) | .Instances[] | select(.LifecycleState == "Terminating:Wait") | .InstanceId')
```

Comment:
Just pushed a commit to address your comments, let me know if there's anything else you'd like to change.